mypy/dmypy_server.py

```diff
@@ -208,19 +208,34 @@ def _response_metadata(self) -> dict[str, str]:
 
     def serve(self) -> None:
         """Serve requests, synchronously (no thread or fork)."""
+
+        class WriteToConn:
```

Comment:
I think mypyc doesn't support classes within functions (or at least not fully supports them).

Reply:
Thanks! I will move it out. I am also not sure it implements all the required methods to replace stdout/stderr.